scx.full: 1.0.14 -> 1.0.15

Gliczy opened the pull request:
https://github.com/sched-ext/scx/releases/tag/v1.0.15
https://github.com/sched-ext/scx/compare/v1.0.14...v1.0.15

Removed the `libalpm` flag since it has been removed upstream.
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Gliczy committed:
scx.full: 1.0.14 -> 1.0.15

Gliczy commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 437169`
Commit: `c24b7133aee2224b37bf353a455a0db731f6d38b`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>scx.cscheds</li>
    <li>scx.cscheds.bin</li>
    <li>scx.cscheds.dev</li>
    <li>scx.full</li>
    <li>scx.full.bin</li>
    <li>scx.full.dev</li>
    <li>scx.rustscheds</li>
  </ul>
</details>

JohnRTitor commented on pkgs/os-specific/linux/scx/scx_rustscheds.nix:
```diff
@@ -47,7 +47,6 @@ rustPlatform.buildRustPackage {
       "-C link-args=-lelf"
       "-C link-args=-lz"
       "-C link-args=-lzstd"
-      "-L ${scx.cscheds.dev}/libbpf/src"
```
https://github.com/sched-ext/scx/blob/059ce35c31c5d7906d2886b0e1c235ef5b6f6da4/meson.build#L347-L349

I picked it up from meson.build, but I wonder why the crash though. 

CC @JakeHillion

JohnRTitor replied:
https://github.com/NixOS/nixpkgs/blob/c24b7133aee2224b37bf353a455a0db731f6d38b/pkgs/os-specific/linux/scx/scx_rustscheds.nix#L40-L43

My question should be to upstream, do we need these as well for buliding only Rust schedulers?

Gliczy replied:
Looks good with `libbpf` vendored with cargo, thanks!

JohnRTitor reviewed (commented):
There's scx_cosmos now, you might want to update the list in the module. 

https://github.com/sched-ext/scx/commit/88c8a3eaf70f3da5bdad9d4738900e99ed7cac87

Gliczy committed:
nixos/scx: add missing schedulers

Gliczy committed:
nixos/scx: inherit pkgs.scx.full.meta maintainers

Gliczy committed:
nixosTests.scx: add production ready schedulers

Gliczy committed:
nixosTests.scx: remove scx_central test
- Not considered production-ready by upstream
- Crashes shortly after being run

Gliczy commented on pkgs/os-specific/linux/scx/scx_rustscheds.nix:
```diff
@@ -63,6 +44,7 @@ rustPlatform.buildRustPackage {
     "--skip=compat::tests::test_struct_has_field"
     "--skip=cpumask"
     "--skip=topology"
+    "--skip=proc_data::tests::test_thread_operations"
```
This is needed to skip a failing test for `scxtop`:
```
---- proc_data::tests::test_thread_operations stdout ----

thread 'proc_data::tests::test_thread_operations' panicked at tools/scxtop/src/proc_data.rs:262:34:
called `Result::unwrap()` on an `Err` value: File not found: /proc/44579/task/44656/stat
```

JohnRTitor commented on pkgs/os-specific/linux/scx/scx_rustscheds.nix:
```diff
@@ -37,18 +30,6 @@ rustPlatform.buildRustPackage {
-      "-I${scx.cscheds.dev}/libbpf/include/uapi"
-      "-I${scx.cscheds.dev}/libbpf/include/linux"
-    ];
-    RUSTFLAGS = lib.concatStringsSep " " [
-      "-C relocation-model=pic"
-      "-C link-args=-lelf"
-      "-C link-args=-lz"
-      "-C link-args=-lzstd"
```
These (except the libbpf linking flag) should be kept I think.

JohnRTitor committed:
scx.rustscheds: restore RUSTFLAGS

Per https://github.com/sched-ext/scx/blob/059ce35c31c5d7906d2886b0e1c235ef5b6f6da4/meson.build#L347-L349
but linking libbpf is not required, built in rust one should work fine.

See https://github.com/NixOS/nixpkgs/pull/437169/commits/8ec509cc139a8d5e665690fac7794ee277233f4a#r2302122465 and https://github.com/sched-ext/scx/pull/2592

JohnRTitor reviewed (approved):
LGTM, tested

JohnRTitor commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 437169`

Logs: https://github.com/JohnRTitor/nixpkgs-review-gha/actions/runs/17289203624


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 test built:</summary>
  <ul>
    <li>nixosTests.scx</li>
  </ul>
</details>
